SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestReplicationConditions/SpatialTestReplicationConditions.cpp

```diff
@@ -0,0 +1,604 @@
+		if (!AssertTrue(TestActor_PhysicsEnabled != nullptr, TEXT("Failed to spawn TestActor_PhysicsEnabled")))
+		{
+			return;
+		}
+
+		TestActor_PhysicsDisabled = GetWorld()->SpawnActor<ATestReplicationConditionsActor_Physics>(
+			ActorSpawnPosition, FRotator::ZeroRotator, FActorSpawnParameters());
+		if (!AssertTrue(TestActor_PhysicsDisabled != nullptr, TEXT("Failed to spawn TestActor_PhysicsDisabled")))
+		{
+			return;
+		}
+
+		TestActor_Common->SpawnDynamicComponents();
+		TestActor_CustomEnabled->SpawnDynamicComponents();
+		TestActor_CustomDisabled->SpawnDynamicComponents();
+		TestActor_AutonomousOnly->SpawnDynamicComponents();
+		TestActor_PhysicsEnabled->SpawnDynamicComponents();
+		TestActor_PhysicsDisabled->SpawnDynamicComponents();
+
+		TestActor_CustomEnabled->SetCustomReplicationEnabled(true);
+		TestActor_CustomDisabled->SetCustomReplicationEnabled(false);
+
+		TestActor_PhysicsEnabled->SetPhysicsEnabled(true);
+		TestActor_PhysicsDisabled->SetPhysicsEnabled(false);
+
+		const bool bWrite = true;
+		bool bCondIgnore[COND_Max]{};
+		ProcessCommonActorProperties(bWrite, bCondIgnore);
+
+		ProcessCustomActorProperties(TestActor_CustomEnabled, bWrite, /*bCustomEnabled*/ true);
+		ProcessCustomActorProperties(TestActor_CustomDisabled, bWrite, /*bCustomEnabled*/ false);
+
+		ProcessAutonomousOnlyActorProperties(bWrite, /*bAutonomousExpected*/ false, /*bSimulatedExpected*/ false);
+
+		ProcessPhysicsActorProperties(TestActor_PhysicsEnabled, bWrite, /*bPhysicsEnabled*/ true, /*bPhysicsExpected*/ true);
+		ProcessPhysicsActorProperties(TestActor_PhysicsDisabled, bWrite, /*bPhysicsEnabled*/ false, /*bPhysicsExpected*/ false);
+
+		AController* PlayerController = Cast<AController>(GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1)->GetOwner());
+		if (!AssertTrue(PlayerController != nullptr, TEXT("Failed to retrieve player controller")))
+		{
+			return;
+		}
+
+		AssertTrue(PlayerController->HasAuthority(), TEXT("Server 1 requires authority over controller"));
+		TestActor_Common->SetOwner(PlayerController);
+
+		TestActor_AutonomousOnly->SetOwner(PlayerController);
+		TestActor_AutonomousOnly->SetAutonomousProxy(true);
+
+		// Set both physics actor to autonomous so properties won't be replicated through simulated condition
+		TestActor_PhysicsEnabled->SetOwner(PlayerController);
+		TestActor_PhysicsDisabled->SetOwner(PlayerController);
+		TestActor_PhysicsEnabled->SetAutonomousProxy(true);
+		TestActor_PhysicsDisabled->SetAutonomousProxy(true);
+
+		RegisterAutoDestroyActor(TestActor_Common);
+		RegisterAutoDestroyActor(TestActor_CustomEnabled);
+		RegisterAutoDestroyActor(TestActor_CustomDisabled);
+		RegisterAutoDestroyActor(TestActor_AutonomousOnly);
+		RegisterAutoDestroyActor(TestActor_PhysicsEnabled);
+		RegisterAutoDestroyActor(TestActor_PhysicsDisabled);
+
+		FinishStep();
+	});
+
+	// Check on non-auth server that actor exists and properties are correct
+	if (bSpatialEnabled)
```

Comment:
if you are adding a step based on it, then I was wrong with recommending the move to the universal step.
This one will have to be determined in the prepare step and stay constant for the lifetime of the actor.